src/ProductConstructionService/ProductConstructionService.Api/Queue/JobsProcessor.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Options;
+using ProductConstructionService.Api.Queue.Jobs;
+
+namespace ProductConstructionService.Api.Queue;
+
+public class JobsProcessor(
+    ILogger<JobsProcessor> logger,
+    IOptions<JobProcessorOptions> options,
+    JobsProcessorScopeManager scopeManager,
+    QueueServiceClient queueServiceClient)
+    : BackgroundService
+{
+    private readonly ILogger<JobsProcessor> _logger = logger;
+    private readonly IOptions<JobProcessorOptions> _options = options;
+    private readonly JobsProcessorScopeManager _scopeManager = scopeManager;
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        await ProcessJobsAsync(cancellationToken);
+    }
```

Comment:
Why do we have this method? Maybe we can merge it with `ProcessJobsAsync`?

Reply:
Moved everything into ExecuteAsync